ArduCopter/mode_systemid.cpp

```diff
@@ -296,6 +407,15 @@ void ModeSystemId::log_data() const
     // Full rate logging of attitude, rate and pid loops
     copter.Log_Write_Attitude();
     copter.Log_Write_PIDS();
+
+    if ((AxisType)axis.get() == AxisType::DISTURB_POS_LAT || (AxisType)axis.get() == AxisType::DISTURB_POS_LONG 
+         || (AxisType)axis.get() == AxisType::DISTURB_VEL_LAT || (AxisType)axis.get() == AxisType::DISTURB_VEL_LONG
+         || (AxisType)axis.get() == AxisType::INPUT_LOITER_LAT || (AxisType)axis.get() == AxisType::INPUT_LOITER_LONG) {
+        pos_control->write_log();
+        copter.logger.Write_PID(LOG_PIDN_MSG, pos_control->get_vel_xy_pid().get_pid_info_x());
+        copter.logger.Write_PID(LOG_PIDE_MSG, pos_control->get_vel_xy_pid().get_pid_info_y());
```

Comment:
Do we end up with double logs if PID logging is on already?

Reply:
No.  this is independently logged for System ID mode